docs/dev-guide/contributing/branching.rst

```diff
@@ -33,6 +33,18 @@ Each ``x.y`` release will have one corresponding branch called ``x.y-dev``. For
 example, all work for the 2.7.z series of releases gets merged into ``2.7-dev``.
 
 
+Protected Branches
+------------------
+
+Within each repository, the master branch, any branch ending in ``-dev``, and any
+branch ending in ``-release`` should be marked as "protected" on GitHub. There is
+a script at
+`devel/scripts/protect-branches.py
+<https://github.com/pulp/devel/tree/master/scripts/protected-branches.py>`_
```

Comment:
I think an additional sentence on which protections we use is important. We disallow force-push and delete, but we don't do anything with required checks.